### PR TITLE
Make WENO-Z default advection scheme

### PR DESCRIPTION
**Pull request by mbkuhn**
Please check the type of change introduced:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): new default to align with common practice

## Checklist

The following is included:

- [ ] new unit-test(s)
- [ ] new regression test(s)
- [ ] documentation for new capability

This PR was tested by running:

- the unit tests
  - [ ] on GPU 
  - [ ] on CPU 
- the regression tests
  - [ ] on GPU 
  - [ ] on CPU

### CODE REVIEW

**Comment by mbkuhn** [comment]
Haven't checked the tests yet; there may need to be some new input lines

**Comment by moprak-nrel** [comment]
Should we change this too?
https://github.com/Exawind/amr-wind/blob/ee0ea504a5a96633e1417b2536176cda44d607a3/amr-wind/equation_systems/AdvOp_Godunov.H#L67-L73

**Comment by marchdf** [comment]
Not including Prakash's comment, here's a list of tests that failed and would probably need an updated input line in their input files. Or we permanently switch some of these to weno (though probably not all).

```
The following tests FAILED:
          2 - abl_godunov (Failed)
          4 - boussinesq_bubble_godunov (Failed)
          5 - freestream_godunov (Failed)
          6 - tgv_godunov (Failed)
          7 - abl_godunov_mpl (Failed)
          8 - abl_godunov_mpl_amr (Failed)
          9 - abl_godunov_cn (Failed)
         10 - abl_godunov_explicit (Failed)
         13 - abl_godunov_noncons (Failed)
         15 - abl_godunov_static_refinement (Failed)
         16 - abl_godunov_static_refinement_rr4 (Failed)
         17 - abl_godunov_scalar_velocity_solve (Failed)
         18 - abl_godunov_segregated_velocity_solve (Failed)
         19 - abl_godunov_timetable (Failed)
         20 - abl_godunov_geostrophic_timetable (Failed)
         21 - abl_ksgsm84_godunov (Failed)
         24 - abl_sampling (Failed)
         34 - act_fixed_wing (Failed)
         35 - act_fixed_wing_fllc (Failed)
         36 - act_moving_wing (Failed)
         37 - act_pitching_wing_2D (Failed)
         38 - act_flat_plate (Failed)
         44 - channel_kwsst (Failed)
         45 - channel_kwsst_sust (Failed)
         46 - channel_kwsstiddes (Failed)
         51 - ekman_spiral (Failed)
         52 - rayleigh_taylor_godunov (Failed)
         67 - abl_bndry_output_native (Failed)
         68 - abl_bndry_output_amr_native (Failed)
         80 - linear_godunov_amr (Failed)
         83 - vortex_dipole_wall_collision (Failed)
         85 - abl_godunov_rayleigh_damping (Failed)
         86 - rankine (Failed)
         87 - rankine-sym (Failed)
         88 - box_refinement (Failed)
         95 - abl_bndry_output (Failed)
         96 - abl_bndry_output_amr_inflow (Failed)
         97 - abl_virtual_lidar (Failed)
         98 - abl_meso_input_dpa (Failed)
         99 - abl_meso_input_ipa (Failed)
        101 - abl_sampling_netcdf (Failed)
        102 - mms_godunov (Failed)
        106 - abl_godunov_hypre (Failed)
        107 - channel_kwsst_hypre (Failed)
        115 - abl_bndry_input_native (Not Run)
        116 - abl_godunov_restart (Not Run)
        117 - abl_bndry_input_amr_native (Not Run)
        118 - abl_bndry_input_amr_native_mlbc (Not Run)
        119 - abl_godunov_forcetimetable (Not Run)
        120 - abl_bndry_input (Not Run)
        121 - abl_bndry_input_amr (Not Run)
        122 - abl_bndry_input_amr_inflow (Not Run)
```

File: amr-wind/equation_systems/icns/icns_advection.H [hunk 1]
```diff
@@ -106,6 +106,9 @@ struct AdvectionOp<ICNS, fvm::Godunov>
             godunov_scheme = godunov::scheme::PPM;
         } else if (amrex::toLower(godunov_type) == "ppm_nolim") {
             godunov_scheme = godunov::scheme::PPM_NOLIM;
+            amrex::Print() << "WARNING: Using advection type ppm_nolim is not "
+                              "recommended. Prefer using wenoz."
```
Comment (marchdf) [inline amr-wind/equation_systems/icns/icns_advection.H:110]:
```suggestion
                              "recommended. Prefer using weno_z."
```

**Comment by mbkuhn** [comment]
On the reg tests, if the test is unrelated to ppm functionality and could be used as a generic example, I would prefer it to use the new default. Or if it's used for verification or validation. However, if it's testing something unrelated to the flow or is a more obscure case that is unlikely to get used, I think we can add the ppm line. E.g., more likely that someone would copy-past specific lines from that input file rather than reuse the whole thing.

Using this reasoning, here's a breakdown for the failed tests (not considering any additional ones from making weno-z default for scalars, too):
 - abl_godunov:                             _ use new default
 - boussinesq_bubble_godunov:  + add ppm line
 - freestream_godunov:                + add ppm line
 - tgv_godunov:                             _ use new default
 - abl_godunov_mpl:                     + add ppm line
 - abl_godunov_mpl_amr:             + add ppm line
 - abl_godunov_cn:                        + add ppm line
 - abl_godunov_explicit:                + add ppm line
 - abl_godunov_noncons:              + add ppm line
 - abl_godunov_static_refinement:                 _ use new default
 - abl_godunov_static_refinement_rr4:          _ use new default
 - abl_godunov_scalar_velocity_solve:           + add ppm line
 - abl_godunov_segregated_velocity_solve:  + add ppm line
 - abl_godunov_timetable:                              _ use new default
 - abl_godunov_geostrophic_timetable:        _ use new default
 - abl_ksgsm84_godunov:            _ use new default
 - abl_sampling:                            + add ppm line
 - act_fixed_wing:                         _ use new default 
 - act_fixed_wing_fllc:                  _ use new default
 - act_moving_wing:                     _ use new default
 - act_pitching_wing_2D:             _ use new default
 - act_flat_plate:                           _ use new default
 - channel_kwsst:                         _ use new default
 - channel_kwsst_sust:                _ use new default
 - channel_kwsstiddes:                _ use new default
 - ekman_spiral:                           _ use new default
 - rayleigh_taylor_godunov:         _ use new default
 - abl_bndry_output_native:         _ use new default
 - abl_bndry_output_amr_native: _ use new default
 - linear_godunov_amr:                 + add ppm line
 - vortex_dipole_wall_collision:     _ use new default
 - abl_godunov_rayleigh_damping: _ use new default
 - rankine:                                      _ use new default
 - rankine-sym:                              _ use new default
 - box_refinement:                        + add ppm line
 - abl_bndry_output:                     _ use new default
 - abl_bndry_output_amr_inflow: _ use new default
 - abl_virtual_lidar:                       + add ppm line
 - abl_meso_input_dpa:               + add ppm line
 - abl_meso_input_ipa:                + add ppm line
 - abl_sampling_netcdf:               + add ppm line
 - mms_godunov:                         _ use new default
 - abl_godunov_hypre:                 _ use new default
 - channel_kwsst_hypre:              _ use new default
 - abl_bndry_input_native:           _ use new default
 - abl_godunov_restart:                _ use new default
 - abl_bndry_input_amr_native:   _ use new default
 - abl_bndry_input_amr_native_mlbc: _ use new default
 - abl_godunov_forcetimetable:    _ use new default
 - abl_bndry_input:                        _ use new default
 - abl_bndry_input_amr:               _ use new default
 - abl_bndry_input_amr_inflow:   _ use new default

Any objections?

**Comment by marchdf** [comment]
I like it

**Comment by moprak-nrel** [comment]
I like the idea of reblessing the golds with the new default, and your reasoning. No objections from my end.

**Comment by mbkuhn** [comment]
What do you think of making a new "abl_godunov_ppm" reg test and removing the "abl_godunov_wenoz" test, as it's effectively replaced by the "abl_godunov" test?

Right now we have abl_godunov, abl_godunov_plm, abl_godunov_nolim, abl_godunov_weno, and abl_godunov_wenoz to cover the fully godunov schemes.

**Comment by mbkuhn** [comment]
@marchdf do you want to check which ones will change now that weno-z is default for scalars, or should we just leave those to the default anyway?

**Comment by marchdf** [comment]
With your latest:
```
The following tests FAILED:
          2 - abl_godunov (Failed)
          6 - tgv_godunov (Failed)
         15 - abl_godunov_static_refinement (Failed)
         16 - abl_godunov_static_refinement_rr4 (Failed)
         19 - abl_godunov_timetable (Failed)
         20 - abl_godunov_geostrophic_timetable (Failed)
         21 - abl_ksgsm84_godunov (Failed)
         34 - act_fixed_wing (Failed)
         35 - act_fixed_wing_fllc (Failed)
         36 - act_moving_wing (Failed)
         37 - act_pitching_wing_2D (Failed)
         38 - act_flat_plate (Failed)
         44 - channel_kwsst (Failed)
         45 - channel_kwsst_sust (Failed)
         46 - channel_kwsstiddes (Failed)
         51 - ekman_spiral (Failed)
         52 - rayleigh_taylor_godunov (Failed)
         60 - abl_godunov_ppm (Failed)
         67 - abl_bndry_output_native (Failed)
         68 - abl_bndry_output_amr_native (Failed)
         83 - vortex_dipole_wall_collision (Failed)
         85 - abl_godunov_rayleigh_damping (Failed)
         86 - rankine (Failed)
         87 - rankine-sym (Failed)
         95 - abl_bndry_output (Failed)
         96 - abl_bndry_output_amr_inflow (Failed)
        102 - mms_godunov (Failed)
        106 - abl_godunov_hypre (Failed)
        107 - channel_kwsst_hypre (Failed)
        115 - abl_bndry_input_native (Not Run)
        116 - abl_godunov_restart (Not Run)
        117 - abl_bndry_input_amr_native (Not Run)
        118 - abl_bndry_input_amr_native_mlbc (Not Run)
        119 - abl_godunov_forcetimetable (Not Run)
        120 - abl_bndry_input (Not Run)
        121 - abl_bndry_input_amr (Not Run)
        122 - abl_bndry_input_amr_inflow (Not Run)
```

**Comment by mbkuhn** [comment]
Cool. The failing tests match the ones in the proposed list using the new default, + abl_godunov_ppm. I think this PR is good to go.

**Review by moprak-nrel — APPROVED** [review]

**Review by marchdf — APPROVED** [review]

**Pull request merged**